src/overlays/actors/ovl_Bg_Spot08_Iceblock/z_bg_spot08_iceblock.c

```diff
@@ -26,47 +34,420 @@ const ActorInit Bg_Spot08_Iceblock_InitVars = {
+extern CollisionHeader D_06001904; // Large iceberg spawned in place of Jabu
+extern Gfx D_06002BD0[];           // Hexagonal ice floe
+extern CollisionHeader D_06002FD8; // Hexagonal ice floe
+
+void BgSpot08Iceblock_SetupAction(BgSpot08Iceblock* this, BgSpot08IceblockActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgSpot08Iceblock_InitDynaPoly(BgSpot08Iceblock* this, GlobalContext* globalCtx, CollisionHeader* colHeader,
+                                   DynaPolyMoveFlag flags) {
+    s32 pad;
+    CollisionHeader* dest;
+    s32 pad2;
+
+    dest = NULL;
+    DynaPolyActor_Init(&this->dyna, flags);
+    CollisionHeader_GetVirtual(colHeader, &dest);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, dest);
+    if (this->dyna.bgId == BG_ACTOR_MAX) {
+        // Warning: move BG registration failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_spot08_iceblock.c", 0xD9,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+// Sets params to 0x10 if not in the cases listed.
+void BgSpot08Iceblock_CheckParams(BgSpot08Iceblock* this) {
+    switch (this->dyna.actor.params & 0xFF) {
+        case 0xFF:
+            this->dyna.actor.params = 0x10;
+            break;
+        default:
+            // Error: arg_data setting error
+            osSyncPrintf("Error : arg_data 設定ミスです。(%s %d)(arg_data 0x%04x)\n", "../z_bg_spot08_iceblock.c", 0xF6,
+                         this->dyna.actor.params);
+            this->dyna.actor.params = 0x10;
+            break;
+        case 1:
+        case 4:
+        case 0x10:
+        case 0x11:
+        case 0x12:
+        case 0x14:
+        case 0x20:
+        case 0x23:
+        case 0x24:
+            break;
+    }
+}
+
+void BgSpot08Iceblock_Bobbing(BgSpot08Iceblock* this) {
+    this->bobOffset = (Math_SinS(this->bobPhaseSlow) * 4.0f) + (Math_SinS(this->bobPhaseFast) * 3.0f);
+}
+
+void BgSpot08Iceblock_SinkUnderPlayer(BgSpot08Iceblock* this) {
+    f32 target;
+    f32 step;
+
+    switch (this->dyna.actor.params & 0xF0) {
+        case 0:
+            step = 0.15f;
+            break;
+        case 0x10:
+            step = 0.2f;
+            break;
+        case 0x20:
+            step = 0.4f;
+            break;
+    }
+
+    // Sink under Player's weight if standing on it
+    target = (func_80043548(&this->dyna) ? -4.0f : 0.0f);
+
+    Math_StepToF(&this->sinkOffset, target, step);
+}
+
+void BgSpot08Iceblock_SetWaterline(BgSpot08Iceblock* this) {
+    this->dyna.actor.world.pos.y = this->sinkOffset + this->bobOffset + this->dyna.actor.home.pos.y;
+}
+
+void BgSpot08Iceblock_MultVectorScalar(Vec3f* dest, Vec3f* v, f32 scale) {
+    dest->x = v->x * scale;
+    dest->y = v->y * scale;
+    dest->z = v->z * scale;
+}
+
+// Cross product
+void BgSpot08Iceblock_CrossProduct(Vec3f* dest, Vec3f* v1, Vec3f* v2) {
```

Comment:
some unnecessary comments in this file, wont point them all out

Reply:
I've removed a few, let me know if there's more you think are superfluous.